### PR TITLE
Docs update for new partitioned table syntax

### PR DESCRIPTION
**Pull request by dorinhogea**
Time to make this public.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/472 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>cldeadlock<br>sc_timepart_logicalsc_generated<br>sc_timepart<br>temp_table_truncate_leak</p>

**Review by riverszhang89 — DISMISSED** [review]

File: docs/pages/programming/sql.md [hunk 1]
```diff
@@ -356,6 +356,13 @@ The list of allowed types that the expression in an index be casted to as well
 as the syntax required to define an index on expression using CSC2 schema, can
 be found [here](table_schema.html#indexes-on-expressions).
 
+The table can be partitioned, if ```PARTITIONED BY``` option is present.  This
+semantic was added in verson 8.0.  A partitioned table is a union of table shards
```
Comment (nirbhayc) [inline docs/pages/programming/sql.md:360]: version

File: docs/pages/programming/sql.md [hunk 2]
```diff
@@ -356,6 +356,13 @@ The list of allowed types that the expression in an index be casted to as well
...
+that are accessed as a whole no different that a regular table.  Currently comdb2
+supports a time-based retention and manual retention partititioning.
```
Comment (nirbhayc) [inline docs/pages/programming/sql.md:362]: partitioning

File: docs/pages/programming/table_partition.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+#removing a partition
+DROP TABLE <tablename>
+```
+
+Examples:
+
+Creating a partitioned table containing 7 days of data, with daily rollout.
+
+```
+#creating a time partiton
```
Comment (nirbhayc) [inline docs/pages/programming/table_partition.md:67]: partition

File: docs/pages/programming/table_partition.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+If the client would choose periodicity `daily`, and retention 31, at all time the partition contains between 30 and 31 days worth of data. Every day a rollout occurs in this case. There is a slight overhead of having 31 shards instead of 4 in this case. Alternatively, a client might choose retention to be 5 weeks, in which case there will always be at least 4 weeks worth of data, and no more than 5 weeks.
+
+
+## Main Limitations
+
+The main advantage of partitioning is scalability.  This is achieved by localizing access for each operation to a limited number of shards (ideally the newest one). 
+Therefore, while the partitioned table behaves like any other table, there is no
+unique key constrain across the whole partition.
+Implementing such a constrain would require that every write operation access and lock pages in each shard during constraint validation, effectively denying the main advantage of partitioning, which is locallized access on small(er) btrees.
```
Comment (nirbhayc) [inline docs/pages/programming/table_partition.md:113]: localized

**Review by nirbhayc — DISMISSED** [review]
Pointed some minor typos inline. Looks good otherwise.

**Review by nirbhayc — APPROVED** [review]
Go go go!

**Pull request merged**